docs/infra/example_lightning.py

```diff
@@ -0,0 +1,192 @@
+            
+            # If it has a default in both, compare them
+            if model_default != func_default and func_default is not inspect.Parameter.empty:
+                raise ValueError(f"Field '{field_name}' default value mismatch: model has '{model_default}', function has '{func_default}'.")
+    
+    def model_post_init(self, log__: tp.Any) -> None:
+        """Check that the parameters are compatible with _cls"""
+        super().model_post_init(log__)
+        exca.helpers.validate_kwargs(self._cls, self.dict())
+    
+    def build(self, **kwargs):  #  /!\ **kwargs needed for trainer checkpoint, but bad api for uid?
+        return self._cls(**self.dict(), **kwargs)
+
+
+def args_to_nested_dict(args: list[str]) -> tp.Dict[str, tp.Any]:  # TODO move to exca.helpers?
+    """
+    Parses a list of Bash-style arguments (e.g., --key=value) into a nested dict.
+    """
+    nested_dict = {}
+    for arg in args:
+        # Split argument into key and value
+        key, value = arg.lstrip("--").split("=", 1)
+        # Convert flat key into a nested dictionary
+        keys = key.split(".")
+        current_level = nested_dict
+        for k in keys[:-1]:
+            current_level = current_level.setdefault(k, {})
+        current_level[keys[-1]] = value
+    return nested_dict
+
+
+class ModelConfig(AutoConfig):  # question: right design?
+    pretrained: bool = True
+    learning_rate: float = 0.001
+    _cls = ResNet
+    
+
+class MnistConfig(AutoConfig):
+    batch_size: int = 64  # question: uid change if add new param, but corresponds to default trainer?
+    _cls = Mnist
+
+
+class TrainerConfig(AutoConfig):
+    max_epochs: tp.Optional[int] = None
+    _cls = Trainer
+
+
+class Experiment(pydantic.BaseModel):
+    model: ModelConfig = ModelConfig()
+    data: MnistConfig = MnistConfig()
+    trainer: TrainerConfig = TrainerConfig()
+
+    infra: exca.TaskInfra = exca.TaskInfra()
+
+    def build(self):
+        mnist = self.data.build()
+        model = self.model.build()
+        callbacks = None
+        if self.infra.folder:
+            callbacks = [ModelCheckpoint(
+                dirpath=self.infra.uid_folder() / 'checkpoint',
+                save_top_k=1,
+                monitor="val_loss",
+                mode="min")
+            ]
+        trainer = self.trainer.build(callbacks=callbacks)
+        return mnist, model, trainer
+
+    @infra.apply
+    def fit(self):
+        data_loaders, model, trainer = self.build()
+        # Define the checkpoint directory
+        checkpoint_dir = self.infra.uid_folder() / 'checkpoint'
+
+        # Find the latest checkpoint if it exists
+        checkpoints = sorted(checkpoint_dir.glob('*.ckpt'))
+        ckpt_path = sorted(checkpoints)[-1] if checkpoints else None
+        
+        # Fit model
+        trainer.fit(model, data_loaders, ckpt_path=ckpt_path)
+        return model
```

Comment:
this is the main reason why I'd like to give a go to trainings... returning a model will pickle it, which is usually considered a bad practice, you're usually supposed to only pickle the weights

Reply:
Good point. It's already dumped via ckpt_path, so there no need to return it actually.

Reply:
solved